constructor JSCJavaScriptExecutorFactory in class JSCJavaScriptExecutorFactory cannot be applied to given types

Hello,
Using this lib with the latest stable RN version 0.51.0 I have encoutered the following issue when trying to run an Android project
```
myproject/node_modules/react-native-threads/android/src/main/java/com/reactlibrary/ReactContextBuilder.java:57: error: constructor JSCJavaScriptExecutorFactory in class JSCJavaScriptExecutorFactory cannot be applied to given types;
        JavaScriptExecutor jsExecutor = new JSCJavaScriptExecutorFactory().create();
                                        ^
  required: String,String
  found: no arguments
  reason: actual and formal argument lists differ in length
1 error
:react-native-threads:compileReleaseJavaWithJavac FAILED
```
I have scoured the list of breaking changes since 0.49 but haven't managed to pinpoint the culprit.
I also haven't had the chance to try this on an IOS device.
Could I  trouble you for a quick fix?

Thank you very much.

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -5,7 +5,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.3.1'
+        classpath 'com.android.tools.build:gradle:2.2.3'
     }
 }
 

--- a/android/src/main/java/com/reactlibrary/ReactContextBuilder.java
+++ b/android/src/main/java/com/reactlibrary/ReactContextBuilder.java
@@ -1,6 +1,7 @@
 package com.reactlibrary;
 
 import android.content.Context;
+import android.net.Uri;
 
 import com.facebook.react.NativeModuleRegistryBuilder;
 import com.facebook.react.ReactPackage;
@@ -19,6 +20,8 @@ import com.facebook.soloader.SoLoader;
 
 import java.util.ArrayList;
 import java.util.concurrent.Callable;
+
+import static com.facebook.react.modules.systeminfo.AndroidInfoHelpers.getFriendlyDeviceName;
 
 public class ReactContextBuilder {
 
@@ -54,7 +57,10 @@ public class ReactContextBuilder {
     }
 
     public ReactApplicationContext build() throws Exception {
-        JavaScriptExecutor jsExecutor = new JSCJavaScriptExecutorFactory().create();
+        String appName = Uri.encode(parentContext.getPackageName());
+        String deviceName = Uri.encode(getFriendlyDeviceName());
+        JavaScriptExecutor jsExecutor = new JSCJavaScriptExecutorFactory(appName, deviceName)
+                .create();
 
         // fresh new react context
         final ReactApplicationContext reactContext = new ReactApplicationContext(parentContext);

--- a/package.json
+++ b/package.json
@@ -23,6 +23,6 @@
     "index.js"
   ],
   "peerDependencies": {
-    "react-native": "~0.49.0"
+    "react-native": ">=0.50.0"
   }
 }

